src/template/timings.rs

```diff
@@ -0,0 +1,337 @@
+use std::{collections::HashMap, fs, io::Error, str::FromStr};
+use tinyjson::JsonValue;
+
+use crate::template::Day;
+
+static TIMINGS_FILE_PATH: &str = "./data/timings.json";
+
+/// Represents benchmark times for a single day.
+#[derive(Clone, Debug)]
+pub struct Timing {
+    pub day: Day,
+    pub part_1: Option<String>,
+    pub part_2: Option<String>,
+    pub total_nanos: f64,
+}
+
+/// Represents benchmark times for a set of days.
+/// Can be serialized from / to JSON.
+#[derive(Clone, Debug, Default)]
+pub struct Timings {
+    pub data: Vec<Timing>,
+}
+
+impl Timings {
+    /// Dehydrate timings to a JSON file.
+    pub fn store_file(&self) -> Result<(), Error> {
+        let json = JsonValue::from(self.clone());
+        let mut bytes = vec![];
+        json.format_to(&mut bytes)?;
+        fs::write(TIMINGS_FILE_PATH, bytes)
```

Comment:
```suggestion
        let mut file = fs::File::create(TIMINGS_FILE_PATH)?;
        json.format_to(&mut file)?;
```
I think it's possible to avoid an intermediate buffer.